File: Q4/php/single/Single.php
<?php

class Single
{
    static private $instance;
    //参数
    private $config;

    private function __construct($config)
    {
        $this->config = $config;
        echo "创建实例";
    }

    //防止克隆对象
    private function __clone(){}

    static public function getInstance($config)
    {
        if (!self::$instance instanceof self) {
            self::$instance = new self($config);
        }
        return self::$instance;

    }

    public function getName()
    {
        echo $this->config;
    }
}

File: Q4/python/single/Single.py
class single_instance:
    
    __instance = None
    def __init__(self,config = 0):
        self.__config = config
    
    def __new__(cls, *args, **kwargs):
        if cls.__instance == None:
            cls.__instance = object.__new__(cls, *args, **kwargs)
        return cls.__instance
        
a = single_instance()
b = single_instance()
if a==b:
    print('同一个类实例')